tests/unit/onyxTest.js

```diff
@@ -630,4 +631,95 @@ describe('Onyx', () => {
                 expect(mockCallback.mock.calls[1][0]).toEqual(collectionUpdate);
             });
     });
+
+    it('should not update a subscriber if the value in the cache has not changed at all', () => {
+        const mockCallback = jest.fn();
+        const collectionUpdate = {
+            test_policy_1: {ID: 234, value: 'one'},
+        };
+
+        // Given an Onyx.connect call with waitForCollectionCallback=true
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.COLLECTION.TEST_POLICY,
+            waitForCollectionCallback: true,
+            callback: mockCallback,
+        });
+        return waitForPromisesToResolve()
+            .then(() => {
+                // When merge is called with an updated collection
+                Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1);
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                // Then we expect the callback to have called twice, once for the initial connect call + once for the collection update
+                expect(mockCallback.mock.calls.length).toBe(2);
+
+                // And the value for the second call should be collectionUpdate
+                expect(mockCallback.mock.calls[1][0]).toEqual(collectionUpdate);
+            })
+            .then(() => {
+                // When merge is called again with the same collection not modified
+                Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1);
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                // Then we should not expect another invocation of the callback
+                expect(mockCallback.mock.calls.length).toBe(2);
+            })
+            .then(() => {
+                // WHEN merge is called again with an object of equivalent value but not the same reference
+                Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, _.clone(collectionUpdate.test_policy_1));
```

Comment:
I _think_ this makes sense, that a different reference wouldn't trigger the callback. I've seen so many bugs in the past where code thinks a variable is one reference, but it ends up being another, and there is stale data... It makes me leary, but until it's an actual problem, I think it's OK.

Reply:
Mainly wanted to show here that whether it's a reference or not makes no difference since we are doing a value comparison and not reference comparison. In the previous test, we pass a reference and in the new test we pass a new object. Both have the same value so both should not trigger any data to be set or updates to occur.

>I've seen so many bugs in the past where code thinks a variable is one reference, but it ends up being another, and there is stale data... It makes me leary, but until it's an actual problem, I think it's OK.

I don't think we have any problems like this or I have not observed any at least. I am pretty confident we are always creating new object references before saving them to the cache (at least when using the `Onyx.merge()` method)

e.g.

https://github.com/Expensify/react-native-onyx/blob/0846e16fa97582c245fbdd966b37ce6d32e50465/lib/Onyx.js#L766